### PR TITLE
go-jira: init at 1.0.17

### PR DESCRIPTION
**Pull request by carlosdagos**
###### Motivation for this change

I borked up #26240 (sorry, I clearly need to learn how to git) where I didn't list any motivations. It's just a nice-enough program.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Additionally, this is an updated version to what was intended in the previous PR at #26240.

### CODE REVIEW

File: pkgs/applications/misc/go-jira/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# This file was generated by https://github.com/kamilchm/go2nix v1.2.1
+{ stdenv, buildGoPackage, fetchgit, fetchhg, fetchbzr, fetchsvn }:
```
Comment (nlewo) [inline pkgs/applications/misc/go-jira/default.nix]: Could you remove useless attributes?

File: pkgs/applications/misc/go-jira/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  goPackagePath = "gopkg.in/Netflix-Skunkworks/go-jira.v1";
+
+  src = fetchgit {
+    inherit rev;
+    url = "https://gopkg.in/Netflix-Skunkworks/go-jira.v1";
+    sha256 = "0ap5dzikp934mawigmv7ighajr5yxpck2gpnb2cmkc1p6qa6gn4v";
+  };
+
+  meta = {
```
Comment (nlewo) [inline pkgs/applications/misc/go-jira/default.nix]: Would you like to become the maintener of this packet? In this case, could you add yourself to the maintainers list?

File: pkgs/applications/misc/go-jira/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# This file was generated by https://github.com/kamilchm/go2nix v1.2.1
+{ stdenv, buildGoPackage, fetchgit, fetchhg, fetchbzr, fetchsvn }:
+
+buildGoPackage rec {
+  name = "go-jira-v1-${version}";
```
Comment (nlewo) [inline pkgs/applications/misc/go-jira/default.nix]: I think the name should be `go-jira-${version}`.

File: pkgs/applications/misc/go-jira/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  version = "1.0.17";
+  rev = "b98da3612d6ab0f3872b64ff36e6cf93219fb4d4";
+
+  goPackagePath = "gopkg.in/Netflix-Skunkworks/go-jira.v1";
+
+  src = fetchgit {
```
Comment (nlewo) [inline pkgs/applications/misc/go-jira/default.nix]: Could you use the tag as rev? Something such as `rev = "v${version}"`.

**Review by nlewo — CHANGES_REQUESTED** [review]

**Comment by carlosdagos** [comment]
Thanks for reviewing @nlewo. I've added the changes as you suggested and added myself as a maintainer.

File: pkgs/applications/misc/go-jira/default.nix [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# This file was generated by https://github.com/kamilchm/go2nix v1.2.1
+{ stdenv, buildGoPackage, fetchgit }:
+
+buildGoPackage rec {
+  name = "go-jira-${version}";
+  version = "1.0.17";
+  rev = "b98da3612d6ab0f3872b64ff36e6cf93219fb4d4";
```
Comment (nlewo) [inline pkgs/applications/misc/go-jira/default.nix]: `rev` is then useless

Comment (carlosdagos) [reply pkgs/applications/misc/go-jira/default.nix]: Thanks @nlewo :) This has been amended now.

**Comment by nlewo** [comment]
@carlosdagos Thanks. One last thing, can you squash your commits and remove `(using go2nix)` from your commit message?

**Comment by carlosdagos** [comment]
Sure thing @nlewo. All done now :)

**Comment by nlewo** [comment]
@GrahamcOfBorg build go-jira

**Comment by nlewo** [comment]
@carlosdagos Thanks!

**Pull request merged**